diff -u a/Makefile b/Makefile
--- a/Makefile	2022-10-21 16:31:23.000000000 -0400
+++ b/Makefile	2022-10-21 17:59:47.000000000 -0400
@@ -11,9 +11,9 @@
 # CC0 Public Domain Dedication:
 # http://creativecommons.org/publicdomain/zero/1.0/
 
-C_INCLUDES:=$(shell pkg-config --cflags gobject-introspection-1.0)
+C_INCLUDES:=$(shell pkg-config --cflags gobject-introspection-1.0 glib-2.0)
 CFLAGS=$(C_INCLUDES) -g
-LIBS:=$(shell pkg-config --libs gobject-introspection-1.0 gmodule-2.0)
+LIBS:=$(shell pkg-config --libs gobject-introspection-1.0 gmodule-2.0 glib-2.0)
 
 OBJECTS=tut-greeter.o main.o
 SOURCES=tut-greeter.c tut-greeter.h main.c
