diff --color=auto -ru a/cmake/ConfigureChecks.cmake b/cmake/ConfigureChecks.cmake
--- a/cmake/ConfigureChecks.cmake	2022-12-30 19:37:05
+++ b/cmake/ConfigureChecks.cmake	2022-12-30 19:37:49
@@ -46,7 +46,7 @@
     # if not symlinked or passed in as a manual define.
     message("Falling back to pkg-config for (n)curses detection")
     include(FindPkgConfig)
-    pkg_search_module(CURSES REQUIRED ncurses curses)
+    pkg_search_module(CURSES REQUIRED ncursesw ncurses curses)
     set(CURSES_CURSES_LIBRARY ${CURSES_LIBRARIES})
     set(CURSES_LIBRARY ${CURSES_LIBRARIES})
 endif()
